durable-storage/src/merkle_layer/node.rs

```diff
@@ -519,12 +567,16 @@ pub(super) fn set(root: &mut Option<Arc<MavlNode>>, key: &Key, data: Bytes) -> b
         // The key already exists and should be updated.
         Ordering::Equal => {
             let node = Arc::make_mut(node);
+            if node.data == data {
+                return false;
+            }
```

Comment:
This is an optimisation I'm not sure about. The likelihood that data has changed is high, but the cost of checking regardless of whether it has changed could be high as well. Would you mind moving this out of this PR, as it is not directly related to hashing?